### PR TITLE
feat(docs): add section about file structure and module types

### PR DESCRIPTION
**Pull request by DavHau**
@steveeJ  @jost-s

### CODE REVIEW

File: modules/flake-parts/_template.nix [hunk 1]
```diff
@@ -1,14 +1,25 @@
...
     ...
-  }: {};
-  flake = {};
+  }: {
+    # system specific outputs like, apps, checks, packages
+
+    # packages = ...
+  };
+  flake = {
+    # system independent outputs like nixosModules, nixosConfigurations, etc.
```
Comment (steveej) [inline modules/flake-parts/_template.nix:21]: the elaboration here is very helpful

File: README.md [hunk 2]
```diff
@@ -33,7 +33,28 @@ To change the definition of some attribute seen in `nix flake show`, adapt the f
...
+Code shared between hosts should be factored out into one or more nixos modules located under `./modules/nixos/`.
+
+## Module types & File structure
+
+This project splits up its nix code into modules in order to improve maintainability and portability of the code.
+
+There are two kinds of modules:
+
+1. **nixos modules**: located under `./modules/nixos`
+1. **flake-parts modules**: located under `./modules/flake-parts`
```
Comment (jost-s) [inline README.md]:
```suggestion
There are two kinds of modules:
 - **nixos modules**: located under `./modules/nixos`
	 - contain configuration for linux or macos hosts.
 - **flake-parts modules**: located under `./modules/flake-parts`
	 - export entities like packages, apps, or machine configurations via the flake outputs, the projects `public API` so to speak.
	 - are responsible for everything seen in the output of `nix flake show`.
```

**Review by jost-s — APPROVED** [review]

**Review by steveej — APPROVED** [review]
very useful docs, thanks!

i'm happy with the docs as is but i don't mind getting Jost's suggstions in either

File: modules/flake-parts/_template.nix (hunk 1 above)
Comment (DavHau) [reply modules/flake-parts/_template.nix:21]: Applied the suggestion. I think this is ready

**Pull request merged**